src/js/actions/tools.js

```diff
@@ -148,64 +149,73 @@ define(function (require, exports) {
             outset = isArtboard ? inset : 27,
             distortModifier = system.isMac ? { command: true } : { control: true };
 
+        // Rectangles used in polices below
+        var innerRect = {
+                x: psSelectionTL.x + inset,
+                y: psSelectionTL.y + inset,
+                width: Math.max(psSelectionWidth - (inset * 2), 0),
+                height: Math.max(psSelectionHeight - (inset * 2), 0)
+            },
+            middleRect = {
+                x: psSelectionTL.x - inset,
+                y: psSelectionTL.y - inset,
+                width: psSelectionWidth + (inset * 2),
+                height: psSelectionHeight + (inset * 2)
+            },
+            outerRect = {
+                x: psSelectionTL.x - outset,
+                y: psSelectionTL.y - outset,
+                width: psSelectionWidth + (outset * 2),
+                height: psSelectionHeight + (outset * 2)
+            };
+
         var insidePolicy = new PointerEventPolicy(adapterUI.policyAction.NEVER_PROPAGATE,
                 adapterOS.eventKind.LEFT_MOUSE_DOWN,
                 {}, // no modifiers inside
-                {
-                    x: psSelectionTL.x + inset,
-                    y: psSelectionTL.y + inset,
-                    width: Math.max(psSelectionWidth - inset * 2, 0),
-                    height: Math.max(psSelectionHeight - inset * 2, 0)
-                }
+                innerRect
             ),
             insideCommandPolicy = new PointerEventPolicy(adapterUI.policyAction.NEVER_PROPAGATE,
                 adapterOS.eventKind.LEFT_MOUSE_DOWN,
                 distortModifier,
-                {
-                    x: psSelectionTL.x + inset,
-                    y: psSelectionTL.y + inset,
-                    width: Math.max(psSelectionWidth - inset * 2, 0),
-                    height: Math.max(psSelectionHeight - inset * 2, 0)
-                }
+                innerRect
             ),
             // Used for distort/skew transformations
             noOutsetCommandPolicy = new PointerEventPolicy(adapterUI.policyAction.ALPHA_PROPAGATE,
                 adapterOS.eventKind.LEFT_MOUSE_DOWN,
                 distortModifier,
-                {
-                    x: psSelectionTL.x - inset,
-                    y: psSelectionTL.y - inset,
-                    width: psSelectionWidth + inset * 2,
-                    height: psSelectionHeight + inset * 2
-                }
+                middleRect
+            ),
+            // Used for proportional resize
+            noOutsetShiftPolicy = new PointerEventPolicy(adapterUI.policyAction.ALPHA_PROPAGATE,
```

Comment:
This is actually geometrically redundant as a result of the outer alpha shift policy for constrained rotation, but I left it in for documentation reasons.